python/benchmarks/bi_benchmarks.py

```diff
@@ -0,0 +1,168 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+import os
+from pathlib import Path
+from arcticdb import Arctic
+from arcticdb.version_store.processing import QueryBuilder
+
+from .common import *
+
+
+def get_query_groupby_city_count_all(q):
+    return q.groupby("City").agg({"Keyword": "count"})
+
+
+def get_query_groupby_city_count_isin_filter(q):
+    return q[q["Keyword"].isin(["bimbo", "twat", "faggot"])].groupby("City").agg({"Keyword": "count"})
+
+
+def get_query_groupby_city_count_filter_two_aggregations(q):
+    return q[q["Keyword"] == "faggot" ].groupby("City").agg({"Keyword": "count", "Number of Records" : "sum"})  
+
+def assert_frame_equal(pandas_df:pd.DataFrame, arctic_df:pd.DataFrame):
+    arctic_df.sort_index(inplace=True)
+    pd.testing.assert_frame_equal(pandas_df,
+                                    arctic_df, 
+                                    check_column_type=False, 
+                                    check_dtype=False)
+
+
+class BIBenchmarks:
+    '''
+        Sample test benchmark for using one opensource BI CSV source.
+        The logic of a test is 
+            - download if parquet file does not exists source in .bz2 format
+            - convert it to parquet format
+            - prepare library with it containing  several symbols that are constructed based on this DF
+            - for each query we want to benchmark do a pre-check that this query produces SAME result on Pandas and arcticDB
+            - run the benchmark tests
+    '''
+
+
+    number = 2
+    timeout = 6000
+    LIB_NAME = "BI_benchmark_lib"
+    # We use dataframe in this file
+    CITY_BI_FILE = "data/CityMaxCapita_1.csv.bz2"
+    CITY_BI_FILE2 = "data/CityMaxCapita_1.parquet.gzip"
+
+    #Defines how many times bigger the database is
+    params = [1, 10]
+
+    def __init__(self):
+        self.lib_name = BIBenchmarks.LIB_NAME
+        self.symbol = self.lib_name
+
+    def setup_cache(self):
+
+        file = os.path.join(Path(__file__).resolve().parent.parent, BIBenchmarks.CITY_BI_FILE2)
+        if (not os.path.exists(file)) :
```

Comment:
Why is this needed since the file is checked in?

Reply:
Originally I implemented only downloading, but it was slow. Then decided to leave this here, so that in future thise code can be copied and run even without having the data with you